source/common/upstream/health_discovery_service.cc

```diff
@@ -30,22 +39,46 @@ void HdsDelegate::establishNewStream() {
     return;
   }
 
-  sendHealthCheckRequest();
-}
-
-void HdsDelegate::sendHealthCheckRequest() {
-  ENVOY_LOG(debug, "Sending HealthCheckRequest");
+  // TODO(lilika): Add support for other types of healthchecks
+  health_check_request_.mutable_capability()->add_health_check_protocol(
+      envoy::service::discovery::v2::Capability::HTTP);
+  ENVOY_LOG(debug, "Sending HealthCheckRequest {} ", health_check_request_.DebugString());
   stream_->sendMessage(health_check_request_, false);
   stats_.responses_.inc();
 }
 
+// TODO(lilika) : Use jittered backoff as in https://github.com/envoyproxy/envoy/pull/3791
 void HdsDelegate::handleFailure() {
-  ENVOY_LOG(warn, "Load reporter stats stream/connection failure, will retry in {} ms.",
-            RETRY_DELAY_MS);
+  ENVOY_LOG(warn, "HdsDelegate stream/connection failure, will retry in {} ms.", RETRY_DELAY_MS);
   stats_.errors_.inc();
   setRetryTimer();
 }
 
+void HdsDelegate::sendResponse() {
+  envoy::service::discovery::v2::HealthCheckRequestOrEndpointHealthResponse response;
+  for (uint i = 0; i < hds_clusters_.size(); i++) {
+    for (const auto& hosts : hds_clusters_[i]->prioritySet().hostSetsPerPriority()) {
+      for (const auto& host : hosts->hosts()) {
+        auto* endpoint = response.mutable_endpoint_health_response()->add_endpoints_health();
+        auto address = host->address()->asString();
+        endpoint->mutable_endpoint()->mutable_address()->mutable_socket_address()->set_address(
+            address.substr(0, address.find_last_of(":")));
```

Comment:
Can't you use the Address::Instance methods directly to get at this information, rather than parsing a string?